### ISSUE TITLE
v9 of SendGrid is available now

### ISSUE THREAD

**Issue by xt0rted** [Azure/azure-webjobs-sdk-extensions#191]
In #170/#171 the NuSpec was updated to limit [SendGrid](https://www.nuget.org/packages/Sendgrid/) to v8 because of it's fully dynamic api. This month they've released v9 which removes all the dynamic stuff with a properly typed & async api (like v7 had 🙄). This is another breaking change between versions so we can't use it with the current SendGrid binding.

My way around dealing with v8 was switching to the SendGrid binding so I haven't tried using the new version to see how it compares functionally to v8.

**Comment by chadwackerman** [comment]
Just as a FYI, this is a *terrible* library. It's also an outrageously expensive service. Microsoft jumped into bed with these guys a few years back so I signed up. After watching them fumble through some awful SDK releases I jumped ship.

They're on version nine (9) of a library that... sends emails. Let that sink in for a moment.

And it's a breaking change. Again.

If you're just sending transactional emails, just spend the half hour and set up Amazon SES. You'll thank me.

Also it's not immediately clear why there is branded middleware promoting this company versus just a generic email provider hook in the WebJobs SDK. There shouldn't even be a dependency here.
